src/development/vortex/development/utils/registry.py

```diff
@@ -0,0 +1,163 @@
+from typing import Dict, Any, Union
+from copy import copy
+import inspect
+
+class Registry:
+    def __init__(self, name: str, base_class=None):
+        self.base_class = base_class
+        self.name = name
+        self.modules: Dict[str,Any] = {}
+    
+    def get(self, key):
+        return self.modules[key]
+    
+    def __len__(self):
+        return len(self.modules)
+    
+    def __contains__(self, key: str):
+        return key in self.modules
+    
+    def __repr__(self):
+        cls_name = self.__class__.__name__
+        fmt = f"{cls_name}<{self.name}>"
+        return fmt
+    
+    def __getitem__(self, key: str):
+        """Added to behaves like dict
+
+        Args:
+            key (str): key
+
+        Returns:
+            type: type associated with "key"
+        """
+        return self.get(key)
+    
+    def register_module(self, module, name: Union[str,None]=None, force: bool=False, overwrite: bool=False):
```

Comment:
what do you think about renaming this to just `register`?

Reply:
because `register` is for creating decorator, while this fn only register and allows to be called like normal functions. This allows to pass parameter to decorator

Reply:
if single name is desired, we actually can return decorator or only perform registration based on module,
```
def register(self, module=None, ...):
    if module is None:
        # return binds arguments return decorator
    else:
        # only register
```

Reply:
note that current `register` method return decorator binding `name`, etc to `register_module`,

Reply:
I think it is better to unify them in a single function tbh

Reply:
ok please wait